test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts

```diff
@@ -0,0 +1,378 @@
+        await testUser.getAsset(newToken)?.amountBefore.free!;
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.claim3rdpartyRewards(liqId, newToken),
+        testUser.keyRingPair,
+      );
+
+      await testUser.refreshAmounts(AssetWallet.AFTER);
+      const rewardTokenAfter =
+        await testUser.getAsset(newToken)?.amountAfter.free!;
+
+      expect(rewardTokenBefore).bnEqual(BN_ZERO);
+      expect(rewardTokenAfter).bnGt(BN_ZERO);
+    });
+
+    test("GIVEN promoted pool MGX-Token2,  pool MGX-Token1,  pool MGX-Token3 AND user with activated 3rd party rewards of pool MGX-Token2 for Token1 and Token3 WHEN user claim all rewards  THEN he receive 2 type of rewards", async () => {
+      await createAndPromoteSecondPool(Assets.DEFAULT_AMOUNT.muln(2), true);
+      await Sudo.batchAsSudoFinalized(
+        Assets.mintToken(newToken3, sudo, Assets.DEFAULT_AMOUNT.muln(40e6)),
+        Assets.mintNative(sudo, Assets.DEFAULT_AMOUNT.muln(40e6)),
+        Sudo.sudoAs(
+          sudo,
+          Xyk.createPool(
+            MGA_ASSET_ID,
+            Assets.DEFAULT_AMOUNT.muln(20e6),
+            newToken3,
+            Assets.DEFAULT_AMOUNT.muln(20e6),
+          ),
+        ),
+      );
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      await waitForRewards(testUser, liqId, 20, newToken);
+
+      const liqId3 = await getLiquidityAssetId(MGA_ASSET_ID, newToken3);
+      await Sudo.batchAsSudoFinalized(
+        Sudo.sudoAs(
+          sudo,
+          await ProofOfStake.rewardPool(
+            MGA_ASSET_ID,
+            newToken2,
+            newToken3,
+            Assets.DEFAULT_AMOUNT.muln(10e6),
+            2,
+          ),
+        ),
+        Assets.promotePool(liqId3.toNumber(), 20),
+      );
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken3,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      await waitForRewards(testUser, liqId, 20, newToken3);
+
+      testUser.addAsset(newToken);
+      testUser.addAsset(newToken3);
+      await testUser.refreshAmounts(AssetWallet.BEFORE);
+      const rewardToken1Before =
+        await testUser.getAsset(newToken)?.amountBefore.free!;
+      const rewardToken3Before =
+        await testUser.getAsset(newToken3)?.amountBefore.free!;
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.claim3rdpartyRewards(liqId, newToken),
```

Comment:
should not we use the claimAll extrinsic?  I think that is what the test is about, ensure that claimAll claim both token1 and token3 
maybe Im missign sth :thinking:

Reply:
Function `proofOfStake.claimRewardsAll` doesn't work with 3rd party rewards, so we can't use this one